### PR TITLE
feat: table improvements

### PR DESCRIPTION
**Pull request by marcelo-sousa-mh**
#### What does this PR do?

Adds support for `title` attribute on `Table.TextCell`.
This is necessary to deal with responsiveness, since `Table.TextCell` truncates text. 
Titles can be passed down as `title` attr or use the default value of `primary` or `secondary`, if these are strings. The `title` attribute always take precedence, so it's useful when we want the `title` to have a different format than the content presented, for example when we have `primary` or `secondary` as HTML and the title needs to be a simple text.

#### How should this be manually tested?

1. Any `Table.TextCell` should display its content on hover

#### Any background context you want to provide?

In my perspective adding separated titles for `primary` and `secondary` would be confusing, so a single `title` attribute seemed more natural. LMKWYGT.

### CODE REVIEW

**Review by maxweldsouza — COMMENTED** [review]
@marcelo-sousa-mh, I'm not a fan of this approach. Users will have to hover over every truncated cell. Also, some columns might already have Tooltips like the OS columns on our servers page. Can't we wrap the content and make the row taller?
I think we can also replace tables with cards if we need responsiveness.

**Comment by marcelo-sousa-mh** [comment]
> @marcelo-sousa-mh, I'm not a fan of this approach. Users will have to hover over every truncated cell. Also, some columns might already have Tooltips like the OS columns on our servers page. Can't we wrap the content and make the row taller? I think we can also replace tables with cards if we need responsiveness.

This comes more of an amend to what we currently have. Since TextCell already truncates content, title was missing to show users the whole text. 

> Users will have to hover over every truncated cell.

Maybe not. If they want to see all details, they would click on the item and go to the page. 
Title comes as a fallback in case of an, unlike, content overflow.

**Review by maxweldsouza — APPROVED** [review]
Alright, the code looks good.

File: src/Table/index.js [hunk 1]
```diff
@@ -30,7 +42,7 @@ Table.Body.displayName = "TableBody";
 
 Table.HeaderCell = React.forwardRef(({ children, ...props }, ref) => (
   <th
-    tw="px-6 py-2 bg-white text-left text-xs leading-5 font-medium text-accent-five uppercase tracking-wider"
+    tw="px-6 first:pl-0 last:pr-0 py-2 bg-white text-left text-xs leading-5 font-medium text-accent-five uppercase tracking-wider"
```
Comment (maxweldsouza) [inline src/Table/index.js:45]: We don't need this anymore, right?

**Review by maxweldsouza — APPROVED** [review]
I've left a minor comment @marcelo-sousa-mh. The clickable row looks a bit odd to me because there's no space between the text and the hover background on the left.

File: src/Table/index.js (hunk 1 above)
Comment (marcelo-sousa-mh) [reply src/Table/index.js:45]: Actually we did to set the default style,  but thanks for pointing out. I moved this to the same block where we add the optional padding.

**Comment by marcelo-sousa-mh** [comment]
> I've left a minor comment @marcelo-sousa-mh. The clickable row looks a bit odd to me because there's no space between the text and the hover background on the left.

On [Latitude.sh #59](https://github.com/Maxihost/latitude.sh/pull/59) the background hover is not happening anymore because we currently moved from row click to content click in order to use `<Link>` instead of `onClick`. 
If we decide on keep that style we can get rid of the classes in `onClick || isSelectable` conditional.

**Comment by marcelo-sousa-mh** [comment]
@maxweldsouza I renamed the branch so it could reflect better the task we are doing, but it ended up messing this PR. I'll close this and resume on #406

**Pull request closed without merging**